core/Settings/Interfaces/CustomSettingInterface.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Piwik\Settings\Interfaces;
+
+interface CustomSettingInterface
+{
+    public static function getCustomValue(?int $idSite = null);
+}
```

Comment:
Entirely optional until we bump our PHPStan level to 9+ , but would match the other interfaces:

```suggestion
/**
 * @template T of mixed
 */
interface CustomSettingInterface
{
    /**
     * @return T
     */
    public static function getCustomValue(?int $idSite = null);
}
```

And the matching `@implements CustomSettingInterface<whatever-type-is-correct>` in the places using it.

Reply:
Added